server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -332,11 +333,13 @@ public void createLogTable(final String tableName, final String entryTypeName)
             StringUtils.format(
                 "CREATE TABLE %1$s (\n"
                 + "  id %2$s NOT NULL,\n"
-                + "  %4$s_id VARCHAR(255) DEFAULT NULL,\n"
-                + "  log_payload %3$s,\n"
+                + "  %3$s_id VARCHAR(255) DEFAULT NULL,\n"
+                + "  log_payload_id VARCHAR(255) NOT NULL,\n"
+                + "  log_payload_order SMALLINT NOT NULL,\n"
```

Comment:
Does this require SQL upgrades on metadata storage?

Reply:
Oh, yes. I added labels.

Reply:
I don't think you can roll back to a prior version in this proposed implementation. Can you change it so that you can roll back and forward across a version with this patch?

Reply:
Hmm, probably it's possible. But the `tasklogs` table is only used for a single API to get segments generated by a task (https://github.com/druid-io/druid/blob/master/docs/content/design/indexing-service.md#submitting-tasks-and-querying-task-status). I think supporting rolling back might be too complicated for this single usage. Maybe it makes sense to note this incompatible change properly and users should drop and create `tasklogs` table if they want to roll back in the release notes. What do you think?

Reply:
Thanks. If you do that, then we can remove the "incompatible" tag. And let's all take a bit to consider the schema of the new table, so we don't need to do this again :)

Reply:
Updated to use a new table.

I also changed the configuration name for the task audit logs from `druid.metadata.storage.tables.taskLogTable` to `druid.metadata.storage.tables.taskActionsTable` to avoid a case that users accidentally use the same name for the new table with the old one which will cause errors. I think this should be still labeled as 'incompatible' because of this configuration change.